pkg/asset/machines/aws/machines.go

```diff
@@ -75,7 +79,78 @@ func Machines(clusterID string, region string, subnets map[string]string, pool *
 		machines = append(machines, machine)
 	}
 
-	return machines, nil
+	replicas := int32(total)
+	failureDomains := []machinev1.AWSFailureDomain{}
+
+	for _, zone := range mpool.Zones {
+		subnet := subnets[zone]
+		domain := machinev1.AWSFailureDomain{
+			Subnet: &machinev1.AWSResourceReference{
+				Type: machinev1.AWSIDReferenceType,
+				ID:   &subnet,
+			},
+			Placement: machinev1.AWSFailureDomainPlacement{
+				AvailabilityZone: zone,
+			},
+		}
+		if subnet == "" {
+			domain.Subnet.Filters = &[]machinev1.AWSResourceFilter{{
+				Name:   "tag:Name",
+				Values: []string{fmt.Sprintf("%s-private-%s", clusterID, zone)},
+			}}
+		} else {
+			domain.Subnet.ID = pointer.StringPtr(subnet)
+		}
```

Comment:
```suggestion
		subnet := subnets[zone]
		domain := machinev1.AWSFailureDomain{
			Placement: machinev1.AWSFailureDomainPlacement{
				AvailabilityZone: zone,
			},
		}
		if subnet == "" {
			domain.Subnet=&machinev1.AWSResourceReference{
				Type: machinev1.AWSFiltersReferenceType,
				Filters: &[]machinev1.AWSResourceFilter{{
					Name:   "tag:Name",
					Values: []string{fmt.Sprintf("%s-private-%s", clusterID, zone)},
				}}
			}
		} else {
			domain.Subnet = &machinev1.AWSResourceReference{
				Type: machinev1.AWSIDReferenceType,
				ID:   &subnet,
			}
		}
```
I think you need to do this to get the correct value out here